File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/RecordingStatsImpl.kt
package dev.hebei.camerax_android.video

import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiRecordingStatsApi

class RecordingStatsImpl(impl: CameraXImpl) : PigeonApiRecordingStatsApi(impl) {
    override fun audioStats(pigeon_instance: androidx.camera.video.RecordingStats): androidx.camera.video.AudioStats {
        return pigeon_instance.audioStats
    }

    override fun numBytesRecorded(pigeon_instance: androidx.camera.video.RecordingStats): Long {
        return pigeon_instance.numBytesRecorded
    }

    override fun recordedDurationNanos(pigeon_instance: androidx.camera.video.RecordingStats): Long {
        return pigeon_instance.recordedDurationNanos
    }
}

File: camerax_ios/ios/Classes/video/RecordingStatsImpl.swift
//
//  RecordingStatsImpl.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/19.
//

import Foundation

class RecordingStatsImpl: PigeonApiDelegateRecordingStatsApi {
    func audioStats(pigeonApi: PigeonApiRecordingStatsApi, pigeonInstance: RecordingStats) throws -> AudioStats {
        fatalError()
    }
    
    func numBytesRecorded(pigeonApi: PigeonApiRecordingStatsApi, pigeonInstance: RecordingStats) throws -> Int64 {
        fatalError()
    }
    
    func recordedDurationNanos(pigeonApi: PigeonApiRecordingStatsApi, pigeonInstance: RecordingStats) throws -> Int64 {
        fatalError()
    }
}
